mailbox/api/src/test/java/org/apache/james/mailbox/events/GroupContract.java

```diff
@@ -275,12 +275,75 @@ default void allGroupListenersShouldBeExecutedWhenGenericGroups() throws Excepti
             verify(listener1, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
             verify(listener2, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
         }
+
+        @Test
+        default void groupListenerShouldReceiveEventWhenRedeliver() throws Exception {
+            MailboxListener listener = newListener();
+
+            eventBus().register(listener, GROUP_A);
+
+            eventBus().reDeliver(GROUP_A, EVENT).block();
+
+            verify(listener, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
+        }
+
+        @Test
+        default void redeliverShouldNotThrowWhenAGroupListenerFails() throws Exception {
+            MailboxListener listener = newListener();
+            doThrow(new RuntimeException()).when(listener).event(any());
+
+            eventBus().register(listener, GROUP_A);
+
+            assertThatCode(() -> eventBus().reDeliver(GROUP_A, EVENT).block())
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        default void redeliverShouldThrowWhenGroupNotRegistered() {
+            assertThatThrownBy(() -> eventBus().reDeliver(GROUP_A, EVENT).block())
+                .isInstanceOf(GroupRegistrationNotFound.class);
+        }
+
+        @Test
+        default void redeliverShouldThrowAfterGroupIsUnregistered() {
+            MailboxListener listener = newListener();
+
+            eventBus().register(listener, GROUP_A).unregister();
+
+            assertThatThrownBy(() -> eventBus().reDeliver(GROUP_A, EVENT).block())
+                .isInstanceOf(GroupRegistrationNotFound.class);
+        }
+
+        @Test
+        default void redeliverShouldOnlySendEventToDefinedGroup() throws Exception {
```

Comment:
can you add a test case about:
- redeliver should not send events to key registration listeners